lib/net/ssh/gateway.rb

```diff
@@ -118,13 +123,22 @@ def shutdown!
   #   gateway.close(port)
   #
   # If +local_port+ is not specified, the next available port will be used.
+  #
+  # If local_port is zero then the next available local port will automatically
```

Comment:
i don't think we need 0 and nil to have different behaviour. Using the ephemeral port with 0 is the right way even is local_port is nil (not specified), and next_port should be deleted.